proxy/http2/Http2ClientSession.cc

```diff
@@ -326,6 +327,9 @@ Http2ClientSession::main_event_handler(int event, void *edata)
     break;
 
   case VC_EVENT_WRITE_COMPLETE:
+    if (this->get_half_close_flag()) {
```

Comment:
According to the specification, there's no guarantee of the completion. The stream identifier in the GOAWAY frame says that streams which have higher stream id than the one in GOAWAY frame are not processed at all.

https://tools.ietf.org/html/rfc7540#section-6.8
```
The last stream identifier in the GOAWAY frame contains the highest-
numbered stream identifier for which the sender of the GOAWAY frame
might have taken some action on or might yet take action on.  All
streams up to and including the identified stream might have been
processed in some way.  The last stream identifier can be set to 0 if
no streams were processed.
```